### ISSUE TITLE
JVM: Form Data Post throws exception on Android 7 + 8

### ISSUE THREAD

**Issue by jonasbark** [ktorio/ktor#927]
### Ktor Version
`1.1.2`

### Ktor Engine Used(client or server and name)
`io.ktor:ktor-client-okhttp:1.1.2`

### JVM Version, Operating System and Relevant Context
Samsung Galaxy S6 - Android 7.0

### Feedback
This is a copy of the https://github.com/Kotlin/kotlinx-io/issues/35 issue - I think ktor might be the better place for this issue.

A minimum example that crashes Android 7 + Android 8 but works fine on Android 9 (e.g. OnePlus 6T):

```
  val result = HttpClient(OkHttp).submitFormWithBinaryData<String>(formData {
                append(key = "header", value = "test")
            }) {
                url("http://httpbin.org/anything")
            }
```

```
I/System.out: java.lang.NoSuchMethodError: No virtual method getParameterCount()I in class Ljava/lang/reflect/Method; or its super classes (declaration of 'java.lang.reflect.Method' appears in /system/framework/core-oj.jar)
W/System.err: java.lang.NoSuchMethodError: No virtual method getParameterCount()I in class Ljava/lang/reflect/Method; or its super classes (declaration of 'java.lang.reflect.Method' appears in /system/framework/core-oj.jar)
W/System.err:     at kotlinx.coroutines.io.internal.CoroutinesEventLoop$FutureReflectionImpl.<clinit>(EventLoopExperimental.kt:36)
                  at kotlinx.coroutines.io.internal.EventLoopExperimentalKt.<clinit>(EventLoopExperimental.kt:48)
W/System.err:     at kotlinx.coroutines.io.internal.EventLoopExperimentalKt.detectEventLoop(EventLoopExperimental.kt:51)
W/System.err:     at kotlinx.coroutines.io.jvm.javaio.BlockingAdapter.parkingLoop(Blocking.kt:228)
W/System.err:     at kotlinx.coroutines.io.jvm.javaio.BlockingAdapter.submitAndAwait(Blocking.kt:215)
                  at kotlinx.coroutines.io.jvm.javaio.BlockingAdapter.submitAndAwait(Blocking.kt:186)
W/System.err:     at kotlinx.coroutines.io.jvm.javaio.InputAdapter.read(Blocking.kt:55)
                  at okio.Okio$2.read(Okio.java:140)
W/System.err:     at okio.RealBufferedSink.writeAll(RealBufferedSink.java:105)
                  at io.ktor.client.engine.okhttp.StreamRequestBody.writeTo(StreamRequestBody.kt:13)
                  at okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.java:72)
W/System.err:     at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
                  at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:45)
```

**Comment by cy6erGn0m** [comment]
This is fixed in kotlinx-io 0.1.5.

**Comment by cy6erGn0m** [comment]
Fixed in 1.1.3.
